packages/appstore/src/components/wallet-cards-carousel/useCarousel.tsx

```diff
@@ -0,0 +1,234 @@
+import { useState, useCallback, useEffect, useLayoutEffect } from 'react';
+
+export interface WalletCardsCarouselResult {
+    readonly pages: number[];
+    readonly activePageIndex: number;
+    readonly prev: () => void;
+    readonly next: () => void;
+    readonly goTo: (pageIndex: number) => void;
+    readonly scrollRef: (el: HTMLElement | null) => void;
+}
+
+interface WalletCardsCarouselState {
+    readonly pages: number[];
+    readonly activePageIndex: number;
+}
+
+export const useWalletCardCarousel = (): WalletCardsCarouselResult => {
```

Comment:
There are many issues with this hook:
- You are using a ReactElement as a state. You should just use a useRef but even then, there is a better way to build a carousel
- You are manually manipulating based on react element and adding event listeners to this react element, treating it as an actual element in the DOM
- You are adding it in the dependency which will not work as expected, the only reason why it work is because of the other changing state in the useEffect dependency

TLDR: Just use a library and extend if you are missing features. This is bad implementation of a carousel and would lead to problems if people would extend it